src/main/java/org/sonatype/goodies/packageurl/PackageUrl.java

```diff
@@ -277,7 +271,16 @@ private static StringBuilder renderSegments(final StringBuilder buff,
    * Parse package-url from given value.
    */
   public static PackageUrl parse(final String value) {
-    return PackageUrlParser.parse(value);
+    return PackageUrlParser.parse(value, true);
+  }
+
+  /**
+   * Parse package-url from given value.
+   *
+   * That is, will *not* make the corrections defined in the spec to the namespace, to the name or to the qualifier keys.
+   */
+  public static PackageUrl parseAsIs(final String value) {
+    return PackageUrlParser.parse(value, false);
```

Comment:
I don't think we want/need to expose this on the primary interface; folks that need special behavior like ignoring the transformations can use the parser directly.

Reply:
Done